Sources_v3/StreamChat/ChatClient.swift

```diff
@@ -271,54 +275,231 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
     ///
     init(
         config: ChatClientConfig,
+        tokenProvider: _TokenProvider<ExtraData>,
         workerBuilders: [WorkerBuilder],
         eventWorkerBuilders: [EventWorkerBuilder],
-        environment: Environment
+        environment: Environment,
+        didReloadUser: @escaping (Error?) -> Void
     ) {
         self.config = config
+        self.tokenProvider = tokenProvider
         self.environment = environment
         self.workerBuilders = workerBuilders
         self.eventWorkerBuilders = eventWorkerBuilders
-        
-        createBackgroundWorkers()
-        createWebSocketClient()
+
+        loadCurrentUserFromDatabase { userId in
+            self.currentUserId = userId
+            self.createWebSocketClient()
```

Comment:
In this form, calling this on a different queue is not safe. `lazy` variable intialization is not thread safe. However, I assume this will be refactored anyway and we will create the WSClient only when we actually need it.